rts/lra/coordinator/src/main/java/io/narayana/lra/coordinator/domain/model/LongRunningAction.java

```diff
@@ -250,7 +250,38 @@ public boolean restore_state(InputObjectState os, int ot) {
             String s = os.unpackString();
             id = s == null ? null : new URI(s);
             s = os.unpackString();
-            parentId = s == null ? null : new URI(s);
+            if (s == null) {
+                parentId = null;
+            } else {
+                parentId = new URI(s);
+
+                LongRunningAction localParent = lraService.lookupTransaction(parentId);
+
+                // if the parent LRA is in the same JVM as the nested LRA (remark, parents are saved before children)
+                // then it's possible to avoid JAX-RS calls by invoking transaction records directly instead:
+                if (localParent != null) {
+                    // this LRA has a parent that is in-VM, so we can optimise away the JAX-RS calls when moving the
+                    // LRA to an end state:
+                    if (par == null) {
```

Comment:
Thank you very much Mike for working on this issue! Much appreciated! Just a little question, what do you think about merging the 2 if statement (line 262 and 265)?